### PR TITLE
Run's status converted to string data type

### PR DESCRIPTION
**Pull request by mparkhe**
## What changes are proposed in this pull request?
 
- RunInfo.status converted to string. Underlying storage layer is unaffected.
- Search filter ascertains status is string
- RunStatus derives all enum key:values from Proto Buf
 
## How is this patch tested?
 
- [x] unit tests
- [x] manual tests: inspection in UI, programmatic query
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [x] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [x] Python

### How should the PR be classified in the release notes? Choose one:
 
- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by aarondav** [comment]
@mparkhe Can you please confirm that you were able to do manual tests against FileStores and SQLAlchemy stores containing existing run info data? I suspect this change could cause problems deserializing data written from a previous version of MLflow.

File: mlflow/store/file_store.py [hunk 1]
```diff
@@ -48,6 +48,10 @@ def _make_persisted_run_info_dict(run_info):
     run_info_dict = dict(run_info)
     run_info_dict['tags'] = []
     run_info_dict['name'] = ''
+    if 'status' in run_info_dict:
+        run_info_dict['status'] = RunStatus.from_string(run_info.status)
```
Comment (aarondav) [inline mlflow/store/file_store.py]: e.g., for old file store data, this may be a number?

File: mlflow/store/file_store.py [hunk 2]
```diff
@@ -59,6 +63,7 @@ def _read_persisted_run_info_dict(run_info_dict):
     dict_copy = run_info_dict.copy()
     if 'lifecycle_stage' not in dict_copy:
         dict_copy['lifecycle_stage'] = LifecycleStage.ACTIVE
+    dict_copy['status'] = RunStatus.to_string(run_info_dict.get('status', RunStatus.RUNNING))
```
Comment (aarondav) [inline mlflow/store/file_store.py]: Maybe add a comment like below indicating why we're making this change. Also do we need a similar isinstance?

File: tests/store/test_sqlalchemy_store.py [hunk 3]
```diff
@@ -860,19 +858,19 @@ def test_search_attrs(self):
...
 
         # change status for one of the runs
         self.store.update_run_info(r2, RunStatus.FAILED, 300)
 
-        filter_string = "attribute.status = {}".format(RunStatus.RUNNING)
+        filter_string = "attribute.status = '{}'".format(RunStatus.to_string(RunStatus.RUNNING))
         six.assertCountEqual(self, [r1], self._search([e1, e2], filter_string))
 
-        filter_string = "attribute.status = {}".format(RunStatus.FAILED)
+        filter_string = "attribute.status = '{}'".format(RunStatus.to_string(RunStatus.FAILED))
```
Comment (aarondav) [inline tests/store/test_sqlalchemy_store.py]: probably we could've just hard-coded the string "FAILED"?

File: mlflow/store/file_store.py (hunk 2 above)
Comment (mparkhe) [reply mlflow/store/file_store.py]: yes. `FileStore` writes status as enum, where as `SqlAlchemyStore` writes it as a string. This is weird, but the APIs out of these stores are `RunInfo`

File: mlflow/store/file_store.py (hunk 1 above)
Comment (mparkhe) [reply mlflow/store/file_store.py]: yes. `FileStore` writes status as enum, where as `SqlAlchemyStore` writes it as a string. This is weird, but the APIs out of these stores are `RunInfo`

**Comment by mparkhe** [comment]
## Manual testing
## `FileStore`
Data is stored in run's `meta.yaml` and status is recorded as an enum (number). Example
```
mlruns/0//c5f53ff93ab14ae9ba9e1e0ba8e74fc3/meta.yaml:status: 3
mlruns/0//4d266bcc08244e15b013ff13b862b861/meta.yaml:status: 3
mlruns/0//4781288efbef4fedb47f8992b2556a4d/meta.yaml:status: 3
mlruns/0//bfcacb1a31594c429005f5f1b09dc4ac/meta.yaml:status: 3
mlruns/0//c4af6d88feb14befbd7e7a0a1cee9c7b/meta.yaml:status: 3
mlruns/0//b12c44de475147d8be2a2cfaedc813fa/meta.yaml:status: 3
```

With the new code changes, `RunInfo` loads this status up as string

```
>>> pprint.pprint(dict(client().get_run('b12c44de475147d8be2a2cfaedc813fa').info), indent=2)
{ 'artifact_uri': 'file:///Users/mani/databricks/mlflow/mlruns/0/b12c44de475147d8be2a2cfaedc813fa/artifacts',
  'end_time': 1557555419879,
  'experiment_id': '0',
  'lifecycle_stage': 'deleted',
  'run_id': 'b12c44de475147d8be2a2cfaedc813fa',
  'run_uuid': 'b12c44de475147d8be2a2cfaedc813fa',
  'start_time': 1557555419848,
  'status': 'FINISHED',
  'user_id': 'mani'}
```

Logging a new run:
```
>>> pprint.pprint(dict(client().get_run('9345b46824c24915b9d170c081d09df6').info), indent=2)
{ 'artifact_uri': 'file:///Users/mani/databricks/mlflow/mlruns/0/9345b46824c24915b9d170c081d09df6/artifacts',
  'end_time': None,
  'experiment_id': '0',
  'lifecycle_stage': 'active',
  'run_id': '9345b46824c24915b9d170c081d09df6',
  'run_uuid': '9345b46824c24915b9d170c081d09df6',
  'start_time': 1557864378949,
  'status': 'RUNNING',
  'user_id': 'mani'}
```

is written as number.

```
mlruns/0//9345b46824c24915b9d170c081d09df6/meta.yaml:status: 1

```

**Pull request merged**